src/gui/progressbardelegate.cpp

```diff
@@ -64,7 +64,8 @@ void ProgressBarDelegate::paint(QPainter *painter, const QStyleOptionViewItem &o
 
     QStyleOptionProgressBar newopt;
     newopt.initFrom(&m_dummyProgressBar);
-    newopt.rect = option.rect;
+    static_cast<QStyleOption &>(newopt) = static_cast<const QStyleOption &>(option);
```

Comment:
Maybe it is worth to put some short comment for this usual code? and the next line too.

Reply:
This looks like a real workaround, so I'd say it **requires** an explanatory comment in the code.